lib/agents/hermes.js

```diff
@@ -32,7 +32,28 @@ class HermesAgent extends ConsoleAgent {
       this.args.unshift('-non-strict');
     }
 
-    this.args.unshift('-Xintl', '-enable-eval', '-fenable-tdz');
+    // Early Hermes versions had `-fenable-tdz`.
+    // In version 0.10.0, it was replaced with `-Xenable-tdz`.
+    // In version 0.13.0, it was removed.
+    // https://github.com/facebook/hermes/commit/f811a0e75f19ab0da7dcf969a61db894fbb3cc35
+    // https://github.com/facebook/hermes/commit/5e2ddd5ed1c1f9af9b5b51c3ac67857a1477cb56
+    try {
+      const cp = await this.createChildProcess(['-version']);
+      let stdout = '';
+      cp.stdout.on('data', str => { stdout += str; });
+      await new Promise(resolve => { cp.on('close', resolve); });
+      // https://github.com/facebook/hermes/blob/66d3f87a8fd0436634be8a1049a1db9d3aa6e9c0/lib/CompilerDriver/CompilerDriver.cpp#L2164
+      const [_, version] = /Hermes release version: (.*)/.exec(stdout);
+      if (/^0[.][0-9][.]/.test(version)) {
+        this.args.unshift('-fenable-tdz');
+      } else if (/^0[.]1[012][.]/.test(version)) {
+        this.args.unshift('-Xenable-tdz');
+      }
```

Comment:
```suggestion
      const [_, releaseVersion] = /Hermes release version: (.*)/.exec(stdout);
      const [__, bytecodeVersion] = /HBC bytecode version: (.*)/.exec(stdout);
      if (/^0[.][0-9][.]/.test(releaseVersion)) {
        this.args.unshift('-fenable-tdz');
      } else if (/^0[.]1[012][.]/.test(releaseVersion) && /^(9[7-9]|[1-9]\d{2,})/.test(bytecodeVersion)) {
        this.args.unshift('-Xenable-tdz');
      }
```

Reply:
since v0.13.0 `--version` still shows 0.12.0 😄

```console
➜  hermes-cli-darwin-v0.12.0 ./hermes --version
LLVM (http://llvm.org/):
  LLVH version 8.0.0svn
  Optimized build

Hermes JavaScript compiler and Virtual Machine.
  Hermes release version: 0.12.0
  HBC bytecode version: 89

  Features:
    Zip file input
```

```console
➜  hermes-cli-darwin-v0.13.0 ./hermes --version
LLVM (http://llvm.org/):
  LLVH version 8.0.0svn
  Optimized build

Hermes JavaScript compiler and Virtual Machine.
  Hermes release version: 0.12.0
  HBC bytecode version: 96

  Features:
    Unicode RegExp Property Escapes
    Zip file input
```

Reply:
remembering now [some time ago](https://github.com/facebook/hermes/issues/1371#issuecomment-2052065772) suggesting we show the short commit id too 😄

Reply:
```suggestion
      const [_, releaseVersion] = /Hermes release version: (.*)/.exec(stdout);
      const [__, bytecodeVersion] = /HBC bytecode version: (.*)/.exec(stdout);
      if (/^0[.][0-9][.]/.test(releaseVersion)) {
        this.args.unshift('-fenable-tdz');
      } else if (/^0[.]1[012][.]/.test(releaseVersion) && !(+bytecodeVersion >= 96)) {
        this.args.unshift('-Xenable-tdz');
      }
```

Reply:
ah yes i messed up the regex, perfect ^